sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridEvent.cs

```diff
@@ -76,10 +95,34 @@ internal EventGridEvent(EventGridEventInternal eventGridEventInternal)
         public string DataVersion { get; set; }
 
         /// <summary>Gets or sets the event data specific to the event type.</summary>
-        internal object Data { get; set; }
+        internal object JsonSerializable
+        {
+            get
+            {
+                return _jsonSerializable;
+            }
+            set
+            {
+                _jsonSerializable = value;
+                Data = new BinaryData(_jsonSerializable, type: DataSerializationType);
```

Comment:
This is strange. You serialize the value but then don't use it when serializing the event.

Can we make this type similar to the CloudEvent?